refactor: Implement TableView component

kgabryje opened the pull request:
### SUMMARY
This PR introduces a new component for building tables with frontend pagination, sorting and filtering. It's based on react-table library. The goal is to replace usages of Table components from `reactable` library with `TableView`.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

kgabryje committed:
Implement TableView component

kgabryje committed:
Add display name

kgabryje committed:
Add className prop

kgabryje committed:
Fix checking if row is selected

kgabryje committed:
Add prop withPagination

nytai commented on superset-frontend/src/components/ListView/TableView.tsx:
```diff
@@ -0,0 +1,136 @@
+  }
+`;
+
+export const TableView = ({
```
Please add some tests for this component

kgabryje committed:
Change dir structure of data view components

kgabryje committed:
Add unit tests

kgabryje commented:
@nytai I added unit tests, but also changed the structure a bit. I moved `TableView` to a separate dir and extracted `TableCollection` and `Pagination` from `ListView` to a new `dataViewCommon` directory. I'd appreciate your feedback.

nytai reviewed (approved):
LGTM. Thanks for adding tests! I agree, moving TableView out of the scope of ListView makes sense, it's a very different component. I'm still unsure about the dataViewCommon abstraction, we could probably just move those components to be regular top level components (along with CardCollection). Since this is currently blocking other refactors from moving forward I'll go ahead and merge and we can revisit the file structure in later PRs.
